test/common/network/dns_impl_test.cc

```diff
@@ -480,11 +480,12 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, DnsImplTest,
 // development, where segfaults were encountered due to callback invocations on
 // destruction.
 TEST_P(DnsImplTest, DestructPending) {
-  ActiveDnsQuery* query = resolver_->resolve("", DnsLookupFamily::V4Only,
-                                             [&](std::list<DnsResponse>&& results) -> void {
-                                               FAIL();
-                                               UNREFERENCED_PARAMETER(results);
-                                             });
+  ActiveDnsQuery* query = resolver_->resolve(
+      "", DnsLookupFamily::V4Only,
+      [&](DnsResolver::ResolutionStatus, std::list<DnsResponse>&& results) -> void {
```

Comment:
Can you actually add expectations around resolution status in this file where appropriate?

Reply:
Yep! I didn't want to spend too much time creating all of this until we discussed https://github.com/envoyproxy/envoy/pull/10137#discussion_r382861884. Will update